core/domain/story_domain.py

```diff
@@ -195,6 +288,30 @@ def __init__(self, story_nodes):
         """
         self.nodes = story_nodes
 
+    def validate(self):
+        """Validates various properties of the story contents object.
+
+        Raises:
+            ValidationError: One or more attributes of the story contents are
+            invalid.
+        """
+        if not isinstance(self.nodes, list):
+            raise utils.ValidationError(
+                'Expected nodes field to be a list, received %s' % self.nodes)
+
+        for node in self.nodes:
+            if not isinstance(node, StoryNode):
+                raise utils.ValidationError(
+                    'Expected each node to be a StoryNode object, received %s' %
+                    node)
+            for destination_node_id in node.destination_node_ids:
+                if next(
+                        (node for node in self.nodes
+                         if node.id == destination_node_id), None) is None:
+                    raise utils.ValidationError(
+                        'Expected all destination nodes to exist')
```

Comment:
Should we also check that the story nodes form a connected acyclic graph?

Reply:
I added a check to make sure a node doesn't point to itself.
Generally, nodes are unlocked when the prerequisites are acquired, right? So, in the check for cyclic, should that be taken into account also?